internal/handlers/middlewares.go

```diff
@@ -74,3 +80,124 @@ func GzipCompress(h http.HandlerFunc) http.HandlerFunc {
 		h.ServeHTTP(ow, r)
 	}
 }
+
+var (
+	// TODO вынести данные переменные в переменные окружения
+	hashKey  = []byte("very-secret-key")  // Симметричный ключ для подписи
+	blockKey = []byte("a-lot-secret-key") // Симметричный ключ для шифрования
+	sCookie  = securecookie.New(hashKey, blockKey)
+)
+
+// ValidateUserUID - декоратор валидации JWT токена.
+func ValidateUserUID(cookieValue string) (string, bool) {
+	var userUID string
+	err := sCookie.Decode("userUID", cookieValue, &userUID)
+	if err == nil {
+		// Кука существует и проходит проверку, продолжаем выполнение следующего обработчика
+		logrus.Printf("Existing valid user ID: %s", userUID)
+		return userUID, true
+	}
```

Comment:
Стоит инвертировать проверку.
```suggestion
	if err == nil {
	    return "", false
	}
        // Кука существует и проходит проверку, продолжаем выполнение следующего обработчика
        logrus.Printf("Existing valid user ID: %s", userUID)
        return userUID, true
```